File: FoodPlanner/app/src/main/java/com/example/foodplanner/MainActivity.java
package com.example.foodplanner;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    TextView CurrentFragmentName;
    BottomNavigationView bottomNavigationView;
     int yemp1,temp2;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        //yemp1 =   = item.getItemId();
        CurrentFragmentName = findViewById(R.id.CurrentFragmentName);
        bottomNavigationView = findViewById(R.id.bottom_navigation);

        //bottomNavigationView = findViewById(R.id.bottom_navigation);

        //MealsRemoteDataSourceImpl temp = MealsRemoteDataSourceImpl.getInstance(this);

        //temp.searchForMeal(this , "Arrabiata");
        // Load default fragment
        if (savedInstanceState == null) {
            CurrentFragmentName.setText("Home");
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, new Fragment_home()).commit();
        }

           // BottomNavigationView bottomNavigationView = null;
        bottomNavigationView.setOnNavigationItemSelectedListener(new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                Fragment selectedFragment = null;

                // Use if-else instead of switch-case
                if (item.getItemId() == R.id.nav_home) {
                    selectedFragment = new Fragment_home();
                    CurrentFragmentName.setText("Home");
                    Log.i("TAG", "Home Selected");
                } else if (item.getItemId() == R.id.nav_favorite) {
                    selectedFragment = new Fragment_Search();
                    CurrentFragmentName.setText("Search");
                    Log.i("TAG", "Search Selected");
                }
                /*else if (item.getItemId() == R.id.nav_favorite) {
                    selectedFragment = new Fragment_Favorite();
                    CurrentFragmentName.setText("Favorites");
                    Log.i("TAG", "Favorites Selected");
                }
                */

                //assert selectedFragment != null;
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, selectedFragment, "csc").commit();
                return true;
            }
        });



    }

}

File: Android_Development/05_Android_Kotlin/05_Session/databinding/fragments/app/src/main/java/com/example/fragments/MainActivity.kt
package com.example.fragments

import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.fragment.app.FragmentManager

internal interface Communicator {
    fun updateProduct(name: String, price: String, imageUrl: String)
}
class MainActivity : AppCompatActivity() ,Communicator{

    var fgr: FragmentManager = supportFragmentManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

    }



    override fun updateProduct(name: String, price: String, imageUrl: String) {
        //Log.i("TAG", "setText: $str")
        val frag:Fragment2 = fgr.findFragmentById(R.id.fragmentContainerView5) as Fragment2
        frag.updateProductInfo(name , price , imageUrl)
    }

}